File: C++/20_Valid_Parentheses.cpp
class Solution {
public:
    bool isValid(string s) {
        if(s.size() % 2 != 0){
            return false;
        }
        if(s == "") return true;
        if(s[0] == ')' || s[0] == ']' || s[0] == '}') return false;
        vector<char> l;
        for(char i : s){
            if(i == '(' || i == '[' || i == '{'){
                if(i == '('){
                    l.push_back(')');
                } else if(i == '['){
                    l.push_back(']');
                } else if(i == '{'){
                    l.push_back('}');
                }
            } else {
                char u = l.back();
                l.pop_back();
                if(i != u){
                    return false;
                }
            }
        }
        if(l.size() != 0){
            return false;
        }
        return true;
    }
};

File: Python/20_Valid_Parentheses.py
class Solution:
    def isValid(self, s: str) -> bool:
        if len(s) % 2 != 0:
            return False
        if s == '':
            return True
        if s[0] == ')' or s[0] == ']' or s[0] == '}':
            return False
        l = []
        for i in s:
            if i == '(' or i == '[' or i == '{':
                if i == '(':
                    l.append(')')
                if i == '[':
                    l.append(']')
                if i == '{':
                    l.append('}')
            else:
                u = l.pop()
                if i != u:
                    return False
        if l != []:
            return False
        return True